### PR TITLE
exitcode normalize;test=develop

### PR DESCRIPTION
**Pull request by lelelelelez**
### PR types

Others
### PR changes

Others
### Describe

统一退出码
测试结果如下：
编译失败：exitcode=7
py35: https://xly.bce.baidu.com/paddlepaddle/paddle/ipipe/#/detail/1230225/job/1749911
coverage:https://xly.bce.baidu.com/paddlepaddle/paddle/ipipe/#/detail/1230227/job/1749915
cpu_py2:https://xly.bce.baidu.com/paddlepaddle/paddle/ipipe/#/detail/1230224/job/1749909
inference:https://xly.bce.baidu.com/paddlepaddle/paddle/ipipe/#/detail/1231206/job/1751398
单测失败：exitcode=8
py35: https://xly.bce.baidu.com/paddlepaddle/paddle/ipipe/#/detail/1231845/job/1752393
coverage:https://xly.bce.baidu.com/paddlepaddle/paddle/ipipe/#/detail/1231846/job/1752395
需要approval：exitcode=6
cpu_py2:https://xly.bce.baidu.com/paddlepaddle/paddle/ipipe/#/detail/1233187/job/1754337

### CODE REVIEW

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -218,6 +218,7 @@ EOF
     # Disable UNITTEST_USE_VIRTUALENV in docker because
     # docker environment is fully controlled by this script.
     # See /Paddle/CMakeLists.txt, UNITTEST_USE_VIRTUALENV option.
+    set +e
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh:221]: 为什么需要加set +e呢？下同

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -241,7 +242,9 @@ EOF
         -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX:-/paddle/build} \
         -DWITH_GRPC=${grpc_flag} \
         -DWITH_LITE=${WITH_LITE:-OFF}
-
+    if [ "$?" != 0 ];then
+        exit 7;
+    fi
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh:247]:
```
编译失败：exitcode=7
单测失败：exitcode=8
需要approval：exitcode=6
```
这些能不能用一个常量？比较容易看懂代码
```
build_error=7
test_error=8
need_approval=6
exit $build_error ?
```

File: paddle/scripts/paddle_build.sh [hunk 3]
```diff
@@ -1170,11 +1177,20 @@ EOF
...
     make -j ${parallel_number} fluid_lib_dist
+    if [ "$?" != 0 ];then
+        exit 7;
+    fi
     make -j ${parallel_number} inference_lib_dist
+    if [ "$?" != 0 ];then
+        exit 7;
+    fi
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh:1187]: 只留下1191-1193这段的退出码，能获取到前面cmake或者make fluid_lib_dist的错误么？

File: paddle/scripts/paddle_build.sh (hunk 1 above)
Comment (lelelelelez) [reply paddle/scripts/paddle_build.sh:221]: set +e是为了在遇到返回值为非零的情况将会继续执行下面的脚本
若不加该命令，假如在cmake 或make中有非0的情况，会自动退出程序，且退出码为2。

File: paddle/scripts/paddle_build.sh (hunk 2 above)
Comment (lelelelelez) [reply paddle/scripts/paddle_build.sh:247]: done

File: paddle/scripts/paddle_build.sh (hunk 3 above)
Comment (lelelelelez) [reply paddle/scripts/paddle_build.sh:1187]: done

**Review by luotao1 — APPROVED** [review]

**Pull request merged**